File: leetcode/java/1342. Number of Steps to Reduce a Number to Zero.java
class Solution {
    public int numberOfSteps (int num) {
        
        int step = 0;
        while(num != 0) {
            if(num % 2 == 0) {
                num /= 2;
            } else {
                num -= 1;
            }
            
            step++;
        }
        
        return step;
    }
}

File: leetcode/kotlin/1342. Number of Steps to Reduce a Number to Zero.kt
class Solution {
    fun numberOfSteps (num: Int): Int {
        
        var step = 0
        var currentNum = num
        while(currentNum != 0) {
            currentNum = if(currentNum % 2 == 0) (currentNum / 2) else (currentNum - 1)
            step++
        }
        
        return step
    }
}
